### PR TITLE
JAMES-3065 exclude Reactor Mono/Flux from API projects

### PR DESCRIPTION
**Pull request by mbaechler**
I'm currently writing the associated ADR but couldn't push it today.
So let's review the code for now.

### CODE REVIEW

File: server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java [hunk 1]
```diff
@@ -84,4 +84,5 @@ public int read() {
     private static int byteToInt(ByteBuffer buffer) {
         return buffer.get() & 0xff;
     }
+
```
Comment (chibenwa) [inline server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java]: noise

File: mailet/api/pom.xml [hunk 2]
```diff
@@ -53,6 +53,12 @@
         <dependency>
             <groupId>${project.groupId}</groupId>
             <artifactId>james-server-util</artifactId>
+            <exclusions>
```
Comment (rouazana) [inline mailet/api/pom.xml]: do you still need this?

File: server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobStore.java [hunk 3]
```diff
@@ -21,7 +21,7 @@
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
 
-import reactor.core.publisher.Mono;
+import org.reactivestreams.Publisher;
```
Comment (rouazana) [inline server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobStore.java]: not a java.util.concurrent.Flow.Publisher?

**Comment by Arsnael** [comment]
```
[25a36be2d3350432c21951f7debea7fe26a96cc8] [ERROR] COMPILATION ERROR : 
[25a36be2d3350432c21951f7debea7fe26a96cc8] [INFO] -------------------------------------------------------------
[25a36be2d3350432c21951f7debea7fe26a96cc8] [ERROR] /james-project/server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/ObjectStorageBlobStoreContract.java:[42,92] cannot find symbol
[25a36be2d3350432c21951f7debea7fe26a96cc8]   symbol:   method block()
[25a36be2d3350432c21951f7debea7fe26a96cc8]   location: interface org.reactivestreams.Publisher<org.apache.james.blob.api.BlobId>
[25a36be2d3350432c21951f7debea7fe26a96cc8] [ERROR] /james-project/server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/ObjectStorageBlobStoreTest.java:[123,110] cannot find symbol
[25a36be2d3350432c21951f7debea7fe26a96cc8]   symbol:   method block()
[25a36be2d3350432c21951f7debea7fe26a96cc8]   location: interface org.reactivestreams.Publisher<org.apache.james.blob.api.BlobId>
[25a36be2d3350432c21951f7debea7fe26a96cc8] [INFO] Preparing remote bundle org.apache:apache-jar-resource-bundle:1.4
[25a36be2d3350432c21951f7debea7fe26a96cc8] [ERROR] /james-project/server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/ObjectStorageBlobStoreTest.java:[139,110] cannot find symbol
[25a36be2d3350432c21951f7debea7fe26a96cc8]   symbol:   method block()
[25a36be2d3350432c21951f7debea7fe26a96cc8]   location: interface org.reactivestreams.Publisher<org.apache.james.blob.api.BlobId>
[25a36be2d3350432c21951f7debea7fe26a96cc8] [ERROR] /james-project/server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/ObjectStorageBlobStoreTest.java:[154,66] cannot find symbol
[25a36be2d3350432c21951f7debea7fe26a96cc8]   symbol:   method block()
[25a36be2d3350432c21951f7debea7fe26a96cc8]   location: interface org.reactivestreams.Publisher<org.apache.james.blob.api.BlobId>
[25a36be2d3350432c21951f7debea7fe26a96cc8] [ERROR] /james-project/server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/ObjectStorageBlobStoreTest.java:[182,13] cannot find symbol
[25a36be2d3350432c21951f7debea7fe26a96cc8]   symbol:   method subscribeOn(reactor.core.scheduler.Scheduler)
[25a36be2d3350432c21951f7debea7fe26a96cc8]   location: interface org.reactivestreams.Publisher<org.apache.james.blob.api.BlobId>
[25a36be2d3350432c21951f7debea7fe26a96cc8] [ERROR] /james-project/server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/ObjectStorageBlobStoreTest.java:[190,13] cannot find symbol
[25a36be2d3350432c21951f7debea7fe26a96cc8]   symbol:   method subscribeOn(reactor.core.scheduler.Scheduler)
[25a36be2d3350432c21951f7debea7fe26a96cc8]   location: interface org.reactivestreams.Publisher<org.apache.james.blob.api.BlobId>
[25a36be2d3350432c21951f7debea7fe26a96cc8] [ERROR] /james-project/server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/ObjectStorageBlobStoreTest.java:[198,13] cannot find symbol
[25a36be2d3350432c21951f7debea7fe26a96cc8]   symbol:   method subscribeOn(reactor.core.scheduler.Scheduler)
[25a36be2d3350432c21951f7debea7fe26a96cc8]   location: interface org.reactivestreams.Publisher<org.apache.james.blob.api.BlobId>
[25a36be2d3350432c21951f7debea7fe26a96cc8] [INFO] Copying 3 resources from 1 bundle.
[25a36be2d3350432c21951f7debea7fe26a96cc8] [ERROR] /james-project/server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/ObjectStorageBlobStoreTest.java:[204,91] cannot find symbol
[25a36be2d3350432c21951f7debea7fe26a96cc8]   symbol:   method block()
[25a36be2d3350432c21951f7debea7fe26a96cc8]   location: interface org.reactivestreams.Publisher<org.apache.james.blob.api.BlobId>
[25a36be2d3350432c21951f7debea7fe26a96cc8] [ERROR] /james-project/server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/ObjectStorageBlobStoreTest.java:[205,92] cannot find symbol
[25a36be2d3350432c21951f7debea7fe26a96cc8]   symbol:   method subscribeOn(reactor.core.scheduler.Scheduler)
[25a36be2d3350432c21951f7debea7fe26a96cc8]   location: interface org.reactivestreams.Publisher<byte[]>
```

File: server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobType.java [hunk 4]
```diff
@@ -16,15 +16,34 @@
...
+    public BlobType(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public final boolean equals(Object o) {
```
Comment (trantienduchn) [inline server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobType.java]: does it have EqualVerifier test ?

File: server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobStore.java (hunk 3 above)
Comment (mbaechler) [reply server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobStore.java]: Give me Java 9 then (: 
reactivestreams is the api jar that allows to have `Publisher` before Java 9.

File: mailbox/api/src/main/java/org/apache/james/mailbox/quota/MaxQuotaManager.java [hunk 5]
```diff
@@ -121,10 +123,10 @@
     }
 
     default Optional<QuotaSizeLimit> getMaxStorage(Map<Quota.Scope, QuotaSizeLimit> maxStorageDetails) {
-        return OptionalUtils.or(
-            Optional.ofNullable(maxStorageDetails.get(Quota.Scope.User)),
-            Optional.ofNullable(maxStorageDetails.get(Quota.Scope.Domain)),
-            Optional.ofNullable(maxStorageDetails.get(Quota.Scope.Global)));
+        return Stream.of(Quota.Scope.User, Quota.Scope.Domain, Quota.Scope.Global)
```
Comment (nvduc91) [inline mailbox/api/src/main/java/org/apache/james/mailbox/quota/MaxQuotaManager.java]: Maybe in `Quota` class should be convenient method like `allScope()` to make something like `Stream.of(Quota.allScope()`. Is it better readble?

File: server/blob/blob-api/src/main/java/org/apache/james/blob/api/DumbBlobStore.java [hunk 6]
```diff
@@ -98,5 +98,5 @@
      * @return a successful Mono if the bucket is deleted or did not exist
```
Comment (remk) [inline server/blob/blob-api/src/main/java/org/apache/james/blob/api/DumbBlobStore.java]: s/Mono/Publisher ?

**Comment by blackheaven** [comment]
test this please

**Comment by Arsnael** [comment]
Euh... I don't understand why the first commit and its fixups have the name of one of my previous merged PR commit (with its ticket number, JAMES-3057), as it has nothing to do with that here?

**Comment by blackheaven** [comment]
good catch!

rebased for rebase fixes

**Comment by blackheaven** [comment]
no result

test this please

**Comment by Arsnael** [comment]
```
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR] Errors: 
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryBlobStoreTest.deleteBucketConcurrentlyShouldNotFail » NoClassDefFound or...
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryBlobStoreTest.deleteConcurrentlyShouldNotFail » NoClassDefFound org/apac...
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryBlobStoreTest.readBytesShouldNotReadPartiallyWhenDeletingConcurrentlyBigBlob » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryBlobStoreTest.readShouldNotReadPartiallyWhenDeletingConcurrentlyBigBlob » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryBlobStoreTest.saveConcurrentlyWithNonPreExistingBucketShouldNotFail » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.concurrentSaveByteSourceShouldReturnConsistentValues » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.concurrentSaveByteSourceShouldReturnConsistentValues » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.concurrentSaveByteSourceShouldReturnConsistentValues » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.concurrentSaveBytesShouldReturnConsistentValues » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.concurrentSaveBytesShouldReturnConsistentValues » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.concurrentSaveBytesShouldReturnConsistentValues » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.concurrentSaveInputStreamShouldReturnConsistentValues » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.concurrentSaveInputStreamShouldReturnConsistentValues » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.concurrentSaveInputStreamShouldReturnConsistentValues » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.deleteBucketConcurrentlyShouldNotFail » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.deleteConcurrentlyShouldNotFail » NoClassDefFound org/...
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.mixingSaveReadAndDeleteShouldReturnConsistentState » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.readBytesShouldNotReadPartiallyWhenDeletingConcurrentlyBigBlob » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.readShouldNotReadPartiallyWhenDeletingConcurrentlyBigBlob » NoClassDefFound
[e11a185f3de202fe2897a5a4b7611e3703b58686] [ERROR]   MemoryDumbBlobStoreTest.saveConcurrentlyWithNonPreExistingBucketShouldNotFail » NoClassDefFound
```

**Comment by Arsnael** [comment]
```
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR] Errors: 
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   WebAdminServerTest.aSecondRouteWithSameEndpointShouldNotOverridePreviouslyDefinedRoutes:60 » NoClassDefFound
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   WebAdminServerTest.aSecondRouteWithSameEndpointShouldNotOverridePreviouslyDefinedRoutesWhenPublic:83 » NoClassDefFound
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   WebAdminServerTest.privateRoutesShouldBePrioritizedOverPublicRoutes:106 » NoClassDefFound
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   ErrorRoutesTest.setUp:53 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   ErrorRoutesTest.setUp:53 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   ErrorRoutesTest.setUp:53 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   ErrorRoutesTest.setUp:53 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   HealthCheckRoutesTest.setUp:83 » NoClassDefFound Could not initialize class io...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound org/apache/commons/logging/LogFacto...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
[8f39312f382ba630932f7d5917d1c4004ab80400] [ERROR]   TasksRoutesTest.setUp:66 » NoClassDefFound Could not initialize class io.resta...
```

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**